### DISCUSSION TITLE
Is it possible to Introduce logic, to just use previous _MEIXXXXX folders on subsequent launches of a pyinstaller --onefile packaged, python executable?

### DISCUSSION THREAD

**Discussion by Avnsx** [pyinstaller/pyinstaller#7879] [Help] [answered]
Is there a method to create a PyInstaller `--onefile` compiled executable that on its initial run, extracts itself to the `%temp%` directory. Upon subsequent launches, the executable should check if the previously extracted files still exist in `%temp%`. If they do, the executable should run directly from the previous extraction location (in the `_MEIXXXXX` folder), avoiding the need for re-extraction.

In essence, how can I ensure that subsequent launches of my PyInstaller packaged Python executable utilize the existing `_MEIXXXX` folder instead of performing another extraction process?

The reason I am trying to do this, is because I am trying to cut down on making users wait for repetetive extraction times of my pyinstaller executable, every time they click it. Further on I am aware that the one-directory mode would be the most proposed solution to this problem, except for my specific use case this would be very counterproductive in multiple aspects.

**Comment by bwoodsend** [comment, marked as answer]
No and there never will be. It would take just as long for PyInstaller to verify an existing unpacked `_MEIXXXXX` directory (make sure it's exactly the app and version and not corrupted or tampered with) as it would to start from scratch. It's also bad for security to make temporary locations predictable.

**Reply by bersbersbers** [reply]
I do understand that you don't want this for other reason (compare https://github.com/pyinstaller/pyinstaller/issues/7907#issuecomment-1700929304), but I don't follow the arguments your are giving here. In fact, I don't think there is a need to verify an existing unpacked `_MEIXXXXX` directory. In one-directory mode, `pyinstaller` does not verify that either, or does it?

In #7907, I proposed a unique ID for each package which could be used as (part of) the folder name to extract to. I'd propose that if this folder exists, `pyinstaller` should assume it is an exact match of the current executable, and I do not see a big problem with this (again, relative to one-directory mode, that is).

Putting it another way, if you feel running existing code in a user-writable, predictable location is a security issue, I wonder how anyone should use one-directory mode at all. By the way, if you think the answer is "to install in a non-user-writable location (e.g., with admin rights)", I want to counter that many extremely wide-spread software packages (Chrome, VS Code, Git for Windows, Python) offer options to be installed in predictable, user-writable locations just as well.
